packages/@aws-cdk/aws-codedeploy/README.md

```diff
@@ -373,6 +391,218 @@ const application = codedeploy.EcsApplication.fromEcsApplicationName(
+  ),
+  threshold: 1,
+  evaluationPeriods: 1,
+});
+
+const greenApiFailure = new cloudwatch.Alarm(stack, 'Green5xx', {
+  alarmName: stack.stackName + '-Http-5xx-Green',
+  metric: greenTargetGroup.metricHttpCodeTarget(
+    elbv2.HttpCodeTarget.TARGET_5XX_COUNT,
+    { period: cdk.Duration.minutes(1) },
+  ),
+  threshold: 1,
+  evaluationPeriods: 1,
+});
+
+new codedeploy.EcsDeploymentGroup(stack, 'BlueGreenDG', {
+  // CodeDeploy will monitor these alarms during a deployment and automatically roll back
+  alarms: [blueUnhealthyHosts, greenUnhealthyHosts, blueApiFailure, greenApiFailure],
+  autoRollback: {
+    // CodeDeploy will automatically roll back if a deployment is stopped
+    stoppedDeployment: true,
+  },
+  services: [service],
+  blueGreenDeploymentConfig: {
+    blueTargetGroup,
+    greenTargetGroup,
+    listener,
+  },
+  deploymentConfig: codedeploy.EcsDeploymentConfig.CANARY_10PERCENT_5MINUTES,
+});
+```
+
+### Deployment validation and manual deployment approval
+
+CodeDeploy blue-green deployments provide an opportunity to validate the new task definition version running on
+the 'green' ECS task set prior to shifting any production traffic to the new version.  A second 'test' listener
+serving traffic on a different port be added to the load balancer. For example, the test listener can serve
+test traffic on port 9001 while the main listener serves production traffic on port 443.
+During a blue-green deployment, CodeDeploy can then shift 100% of test traffic over to the 'green'
+task set/target group prior to shifting any production traffic during the deployment.
+
+```ts
+declare const myApplication: codedeploy.EcsApplication;
+declare const service: ecs.FargateService;
+declare const blueTargetGroup = elbv2.ITargetGroup;
+declare const greenTargetGroup = elbv2.ITargetGroup;
+declare const listener = elbv2.IApplicationListener;
+declare const testListener = elbv2.IApplicationListener;
+
+new codedeploy.EcsDeploymentGroup(stack, 'BlueGreenDG', {
+  services: [service],
+  blueGreenDeploymentConfig: {
+    blueTargetGroup,
+    greenTargetGroup,
+    listener,
+    testListener,
+  },
+  deploymentConfig: codedeploy.EcsDeploymentConfig.CANARY_10PERCENT_5MINUTES,
+```
+
+Automated validation steps can run during the CodeDeploy deployment after shifting test traffic and before
+shifting production traffic.  CodeDeploy supports registering Lambda functions as lifecycle hooks for
+an ECS deployment.  These Lambda functions can run automated validation steps against the test traffic
+port, for example in response to the `AfterAllowTestTraffic` lifecycle hook.  For more information about
+how to specify the Lambda functions to run for each CodeDeploy lifecycle hook in an ECS deployment, see the
+[AppSpec 'hooks' for an Amazon ECS deployment](https://docs.aws.amazon.com/codedeploy/latest/userguide/reference-appspec-file-structure-hooks.html#appspec-hooks-ecs)
+section in the CodeDeploy user guide.
```

Comment:
When configuring the deployment group through CFN, where do you configure this file? I thought that
this construct replaces the appspec file?

Reply:
No, the appspec file is provided to each deployment, not to the deployment group.  I added instructions to the integ test file that walk through the steps.